crowbar_framework/app/models/api/node.rb

```diff
@@ -72,7 +69,26 @@ def os_upgrade
         Rails.logger.error("Executing of os upgrade script has failed on node #{@node.name}.")
         return false
       end
-      true
+
+      begin
```

Comment:
I've already added a helper. However I do not understand why you keep mentioning `#end_step` here.